docs/naps/3-spaces.md

```diff
@@ -0,0 +1,127 @@
+(nap3)=
+
+# NAP-3 — Spaces
+
+```{eval-rst}
+:Author: Lorenzo Gaifas, brisvag@gmail.com
+:Created: 2022-06-08
+:Resolution: <date resolved, in yyyy-mm-dd format>
+:Status: Draft
+:Type: Standards Track
+:Version effective: <version-number> (for accepted NAPs)
+``` 
+
+
+# Abstract
+
+`napari` is currently limited to holding (and rendering) data belonging to a single, universal space (which we often refer to as *world space*). However, it is often useful to have quick and easy access to different parts of a dataset that do no belong to the same coordinate system. In these cases, not only forcing data to live in the same world makes no sense, but it can make navigating the data and interacting with the viewer slower and less intuitive.
+
+This NAP discusses the reasons why a native napari approach would be better than the currently available workarounds. It then proposes the introduction of `spaces` as a way to manage different coordinate spaces in the same viewer.
+
+
+# Motivation and Scope
+
+This NAP aims to address a few problems that arise (especially with big datasets), when working with many layers that do not necessarily belong to the same *absolute* coordinate space. For example, there is no reason to relate between the *absolute* coordinates of `image 1` and `image 2` from the same microscopy data collection, but it might be useful to quickly switch between the two to -- for example -- compare the effectiveness of a processing step on different images, or to visually inspect qualitative differences.
```

Comment:
It would be good to add what you think is out of scope for this NAP/project - at least I often find that useful to be keep the size of the work manageable.

In particular, I think it would be good to clarify if a layer can be in more than one space. If not, then you don't have to worry so much about the kind of issues discussed in #4682.